clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10969,6 +10969,9 @@ def err_ivdep_declrefexpr_arg : Error<
 def warn_ivdep_redundant : Warning <"ignoring redundant Intel FPGA loop "
   "attribute 'ivdep': safelen %select{INF|%1}0 >= safelen %select{INF|%3}2">,
   InGroup<IgnoredAttributes>;
+def warn_attribute_spelling_deprecated : Warning<
+  "previous spelling of attribute %0 is deprecated">,
```

Comment:
User's aren't particularly knowledgeable about the fact that the same attribute can be spelled in 2 different ways.  They tend to treat them as different attributes with similar/same behavior.

Additionally, we likely want to suggest 

This should probably be:
"attribute %0 is deprecated
Then a note of: <note_function_suggestion > "did you mean %1?"

Additionally, See if you can find if there is a good note that we can use to suggest

Reply:
> User's aren't particularly knowledgeable about the fact that the same attribute can be spelled in 2 different ways. They tend to treat them as different attributes with similar/same behavior.
> 
> Additionally, we likely want to suggest
> 
> This should probably be:
> "attribute %0 is deprecated
> Then a note of: <note_function_suggestion > "did you mean %1?"
> 
> Additionally, See if you can find if there is a good note that we can use to suggest

Thanks @erichkeane  for the suggestion. I will look at this and update the diagnostic.